ospfd/ospf_auth.c

```diff
@@ -222,22 +224,21 @@ static int ospf_auth_check_md5_digest(struct ospf_interface *oi,
 static int ospf_auth_make_md5_digest(struct ospf_interface *oi,
 				     struct ospf_packet *op, struct key *key)
 {
-	void *ibuf;
-	struct ospf_header *ospfh;
+	void *ibuf = STREAM_DATA(op->s);
+	struct ospf_header *ospfh = (struct ospf_header *)ibuf;
 	unsigned char digest[OSPF_AUTH_MD5_SIZE];
-	uint16_t length;
+	unsigned int length = ntohs(ospfh->length);
```

Comment:
uint32_t please in both places

Reply:
I changed all `unsigned int`s to uint32_t. There were more than 2 :)